source/tutorials/packaging/contributing-to-nixpkgs.md

```diff
@@ -0,0 +1,222 @@
+---
+myst:
+  html_meta:
+    "description lang=en": "Contributing to Nixpkgs"
+    "keywords": "Nix, Nixpkgs, contributing"
+---
+
+
+(contributing-to-nixpkgs)=
+
+# Contributing to Nixpkgs
+
+[Nixpkgs](https://github.com/nixos/nixpkgs) is one of the largest public repositories by contributions and *the* largest single collection of software packages available on the web.
+
+In this tutorial, which follows {ref}`the tutorial on packaging existing software <packaging-existing-software>`, you will learn how to prepare newly-packaged software for submission to Nixpkgs, becoming a Nixpkgs contributor in the process.
+
+## Tags, Revisions, and Releases
+An earlier version of the `icat.nix` definition created in the {ref}`Packaging Existing Software <packaging-existing-software>` tutorial used the `master` revision of the [upstream `icat` repository](https://github.com/atextor/icat), which would be suitable for individual use but isn't good practice for submission to Nixpkgs.
+
+It is much better to use a fixed revision corresponding to a particular release version of the software so maintainers (perhaps you!) can easily check whether the package should be updated.
+
+This was already done in the previous tutorial, but as a reminder, here are the steps:
+1. Check the upstream repository for tags or releases. On GitHub, these are available at `https://github.com/<owner>/<repo>/tags` and `https://github.com/<owner>/<repo>/releases` respectively.
+2. Choose the latest tag or release and write it into a string literal as the value of the `rev` attribute passed to the [fetcher](https://nixos.org/manual/nixpkgs/stable/#chap-pkgs-fetchers).
+3. Use `nix-prefetch-url` or a similar tool to fetch, unpack, and hash the files marked by the tag or release you selected.
+4. Alternatively, you can also use a tool such as `nix-init` to automatically generate a required derivation for some common project types, or use its output to help lay down a skeleton implementation of the above steps.
+
+In the `icat` example, the source was hosted on GitHub, so `fetchFromGitHub` was used for the `src` attribute passed to `stdenv.mkDerivation`, with `rev = "v0.5"` corresponding to the latest tag, and a SHA256 hash generated by
+
+```console
+nix-prefetch-url --unpack https://github.com/atextor/icat/archive/refs/tags/v0.5.tar.gz --type sha256`
+```
+
+Here is the final version of `icat.nix` from the previous tutorial:
```

Comment:
I don't think this needs to be repeated. We should assume that users read the previous tutorial.

Reply:
@infinisil I'm not convinced that's a fair assumption, just due to the reality of how people work. However, I do think that we should instead insert a short hyperlink rather than repeating it verbatim.